nntrainer/layers/common_properties.cpp

```diff
@@ -48,6 +48,33 @@ void FilePath::set(const std::string &v) {
 
 std::ifstream::pos_type FilePath::file_size() { return cached_pos_size; }
 
+bool TensorShape::isValid(const std::string &v) const {
+  // Tensor dimension does not allow the 0 & number of dimension shuld be less
+  // than MAXDIM 4. Also the white space will be ignored.
+
+  static std::regex allowed("[1-9:]*");
```

Comment:
"[1-9:]*" -> "[1-9]:[1-9:]*"?
then `!v.empty()` can also be removed from L63.
I wonder if "::" will pass the current checks?

Reply:
fixed.